### PR TITLE
Feature/docs

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR fixes that: resources should be public, all other registries stuff should be private.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/registries/filter.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+INTERNAL_PROTOCOL_ID = PublicId.from_str("fetchai/internal:0.1.0")
+DECISION_MAKER = "decision_maker"
+
+Item = TypeVar("Item")
+ItemId = TypeVar("ItemId")
+ComponentId = Tuple[SkillId, str]
+SkillComponentType = TypeVar("SkillComponentType", Handler, Behaviour, Task, Model)
+
+
+class Filter:
```
Comment (DavidMinarsch) [inline aea/registries/filter.py]: This is not part of the public api, hence now separate from resources

File: aea/registries/base.py [hunk 2]
```diff
@@ -531,297 +521,3 @@ def fetch_internal_handler(self, skill_id: SkillId) -> Optional[Handler]:
         :return: the internal handler registered for the skill id
         """
         return self.fetch_by_protocol_and_skill(INTERNAL_PROTOCOL_ID, skill_id)
-
-
-class Resources:
```
Comment (DavidMinarsch) [inline aea/registries/base.py]: Resources is the only part of the public api in registries/* > removing it from here allows us to expose it in the api docs

File: aea/skills/base.py [hunk 3]
```diff
@@ -559,6 +559,7 @@ def __init__(
         self.handlers = handlers if handlers is not None else {}
         self.behaviours = behaviours if behaviours is not None else {}
         self.models = models if models is not None else {}
+        self.skill_context._skill = self
```
Comment (DavidMinarsch) [inline aea/skills/base.py]: @marcofavorito this seems more consistent as it attaches the skill to the context no matter where the skill is initialized

File: aea/skills/base.py [hunk 4]
```diff
@@ -610,7 +611,7 @@ def from_dir(cls, directory: str, agent_context: AgentContext) -> "Skill":
         skill = Skill(
             skill_config, skill_context, handlers, behaviours, model_instances
         )
-        skill_context._skill = skill
+        # skill_context._skill = skill
```
Comment (DavidMinarsch) [inline aea/skills/base.py]: see above, can be removed imo

File: docs/decision-maker-transaction.md [hunk 5]
```diff
@@ -32,104 +32,106 @@ INPUT_FILE = "input.txt"
...
+```
 
-    identity_2 = Identity(
-        name="my_aea_2",
-        address=wallet_2.addresses.get(FETCHAI),
-        default_address_key=FETCHAI,
+Add a simple skill with a transaction handler.
+
+``` python
+    skill_context = SkillContext(my_aea.context)
```
Comment (DavidMinarsch) [inline docs/decision-maker-transaction.md]: example of programmatically creating a skill. still quite verbose but at least possible with the public api now.

File: tests/test_aea.py [hunk 6]
```diff
@@ -330,10 +328,10 @@ def test_initialize_aea_programmatically(self):
         expected_dummy_task = async_result.get(2.0)
         assert expected_dummy_task.nb_execute_called > 0
 
-        dummy_handler = self.aea.resources.handler_registry.fetch_by_protocol_and_skill(
+        dummy_handler = self.aea.resources.get_handler(
             DefaultMessage.protocol_id, dummy_skill_id
         )
-        dummy_handler_alt = self.aea.resources.handler_registry.fetch(
+        dummy_handler_alt = self.aea.resources._handler_registry.fetch(
```
Comment (DavidMinarsch) [inline tests/test_aea.py]: fetching handlers by name is only done in tests, I don't think we should be making this a recommended behaviour, hence no public api

File: aea/skills/base.py (hunk 3 above)
Comment (marcofavorito) [reply aea/skills/base.py]: ok

File: aea/skills/base.py (hunk 4 above)
Comment (marcofavorito) [reply aea/skills/base.py]: yes

File: tests/test_aea.py [hunk 7]
```diff
@@ -439,10 +435,10 @@ def test_initialize_aea_programmatically(self):
         assert expected_dummy_task.nb_execute_called > 0
 
         dummy_handler_name = "dummy"
-        dummy_handler = self.aea.resources.handler_registry.fetch(
+        dummy_handler = self.aea.resources._handler_registry.fetch(
```
Comment (DavidMinarsch) [inline tests/test_aea.py]: see above

File: tests/test_protocols/test_generator.py [hunk 8]
```diff
@@ -359,7 +355,7 @@ def test_generated_protocol_end_to_end(self):
         agent_1_handler = Agent1Handler(
             skill_context=SkillContext(aea_1.context), name="fake_skill"
         )
-        resources_1.handler_registry.register(
+        resources_1._handler_registry.register(
```
Comment (DavidMinarsch) [inline tests/test_protocols/test_generator.py]: For now I don't think we should add a public API to add handlers or behaviours without a skill. It is not clean given current architecture and comes with potential dangerous side effects (e.g. certain APIs breaking which require skill configs etc present)

File: tests/test_registries.py [hunk 9]
```diff
@@ -180,55 +181,57 @@ def setup_class(cls):
...
 
         # unregister the error handler and test that it has been actually unregistered.
         # TODO shouldn't we prevent the unregistration of this?
-        error_handler = self.resources.handler_registry.fetch(
+        error_handler = self.resources._handler_registry.fetch(
             (self.error_skill_public_id, "error_handler")
         )
         assert error_handler is not None
-        self.resources.handler_registry.unregister(
+        self.resources._handler_registry.unregister(
```
Comment (DavidMinarsch) [inline tests/test_registries.py]: see above

File: tests/test_protocols/test_generator.py (hunk 8 above)
Comment (marcofavorito) [reply tests/test_protocols/test_generator.py]: yes, agreed.

**Pull request merged**